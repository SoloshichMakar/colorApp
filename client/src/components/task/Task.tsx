import React from "react";

import { TaskList, TaskCheckbox, TaskText } from "./style";
import {ITask} from "../../store/types/types";

export interface ITaskProps extends ITask {
    taskIsCompleted: Function;
}

const  Task: React.FC<ITaskProps>  = ({ taskText, id, taskColor, completed, taskIsCompleted }) => {

  const currentTaskIsCompleted = (completedTaskId: string) => {
    taskIsCompleted(completedTaskId, !completed);
  };

  return (
    <TaskList style={{ backgroundColor: completed ? "lightgrey" : taskColor }}>
      <TaskCheckbox
        type="checkbox"
        onClick={() => currentTaskIsCompleted(id)}
        checked={completed}
      />
      <TaskText>{taskText}</TaskText>
    </TaskList>
  );
};
export default Task;
