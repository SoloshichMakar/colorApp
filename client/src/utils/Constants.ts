export const COLORS: Array<string> = [
  "#f16d69",
  "#f279a2",
  "#9170cb",
  "#5eb3f6",
  "#67d7e5",
  "#ffe083",
];

export const SET_COLOR: string = "setColor";
export const ADD_TASK: string = "addTask";
export const ASYNC_ADD_TASK: string = "asyncAddTask";
export const TEXT_CHANGE: string = "textChange";
export const DELETE_TASK: string = "deleteTask";
export const ASYNC_DELETE_TASK: string = "asyncDeleteTask";

export const TASK_IS_DONE: string = "taskIsDone";
export const ASYNC_TASK_IS_DONE: string = "asyncTaskIsDone";

export const GET_DATA: string = 'getData';
export const ASYNC_GET_DATA: string = "asyncGetData";

export const TEXT_EMAIL_CHANGE: string = 'textEmailChange';
export const TEXT_PASSWORD_CHANGE: string = 'textPasswordChange';
export const USER_CHECK: string = 'userCheck';
export const ERROR: string = 'ERROR';
export const ASYNC_USER_CHECK: string = 'asyncUserCheck';
export const CREATE_USER: string = 'createUser';
export const ASYNC_CREATE_USER: string = 'asyncCreateUser';
export const TEXT_CONFIRM_PASSWORD_CHANGE: string = 'textConfirmPasswordChange';
