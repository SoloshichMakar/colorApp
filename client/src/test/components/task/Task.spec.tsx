import React from 'react';
import renderer from 'react-test-renderer';
import { COLORS } from "../../../utils/Constants";

import Task from '../../../components/task/Task';
import {BrowserRouter} from "react-router-dom";


const props = {
    taskText: 'Test Text',
    id: '1',
    taskColor: COLORS[0],
    completed: false,
    taskIsCompleted: Function
};


it('renders correctly', () => {
    const tree = renderer
        .create(
            <BrowserRouter>
                <Task
                    taskText={props.taskText}
                    id={props.id}
                    taskColor={props.taskColor}
                    completed={props.completed}
                    taskIsCompleted={props.taskIsCompleted}
                />
    </BrowserRouter>
)
.toJSON();
    expect(tree).toMatchSnapshot();
});