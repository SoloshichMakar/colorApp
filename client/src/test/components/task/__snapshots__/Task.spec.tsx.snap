// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm sc-AxirZ hameIv"
  style={
    Object {
      "backgroundColor": "#f16d69",
    }
  }
>
  <input
    checked={false}
    className="sc-AxhCb jbKKYH"
    onClick={[Function]}
    type="checkbox"
  />
  <div
    className="sc-AxjAm sc-AxiKw ipfOCv"
  >
    Test Text
  </div>
</div>
`;
